R/RunStep.R

```diff
@@ -4,18 +4,38 @@
-#' step in the workflow. It prepares the parameters for the function call, including the metadata,
-#' mapping, and data inputs. It then calls the function specified in `lStep$name` with the prepared
-#' parameters.
+#' step in the workflow. It prepares the parameters for the function call and then calls the function
+#' specified in `lStep$name` with the prepared parameters.
 #'
-#' @param lStep `list` single workflow step (typically defined in `lWorkflow$workflow`). Should
-#'   include the name of the function to run (`lStep$name`), data inputs (`lStep$inputs`), name of
-#'   output (`lStep$output`) and configurable parameters (`lStep$params`) (if any)
+#' The primary utility of this function is to provide a prioritized parser for function parameterization.
+#' Parameters should be specified as a named list in `lStep$params`, where each element is a key-value pair
+#' that will be parsed and then passed to the specified function as a set of parameter names/values.
+#' Parameter values should be specified as scalar strings. Those values are then pulled from `lMeta` or `lData`
+#' when possible. When no matching `lData` or `lMeta` objects are found, parameter values are passed through as
+#' stings. Note that parsing vectorized parameters is not supported at this time; they are passed directly
+#' as character vectors. To pass a vector or list, we recommend saving it as an object in `lData`.
+#'
+#' Full prioritization for parsing parameters is below:
+#'
+#' 1. If a single parameter value is equal to "lMeta", the the full lMeta object is passed to the function
+#' (for the given paramName).
+#' 2. If a single parameter value is equal to "lData", the full lData object is passed to the function.
+#' 3. If a single parameter value is equal to "lSpec", the full lSpec object is passed to the function.
+#' 4. If a single parameter value is found in names(lMeta), that property is pulled from lMeta (e.g.
+#' lMeta${paramVal}) and passed to the function.
+#' 5. If a single parameter value is found in names(lData), that property is pulled from lData (e.g.
+#' lData${paramVal}) and passed to the function.
+#' 6. Otherwise single parameter value is passed to the function as a string.
+#' 7. If the parameter value is a vector, the vector is passed to the function as a vector or strings.
+#'
+#' @param lStep `list` single workflow step (typically pulled from `lWorkflow$steps`). Should
+#'   include the name of the function to run (`lStep$name`), name of the object where the function result should be saved (`lStep$output`) and configurable parameters (`lStep$params`) (if any)
 #' @param lData `list` a named list of domain level data frames.
+#' @param lSpec `list` a data specification containing required columns. See `vignette("gsm_extensions")`.
 #' @param lMeta `list` a named list of meta data.
 #'
 #' @examples
-#' wf_mapping <- MakeWorkflowList("mapping")
+#' wf_mapping <- MakeWorkflowList(strPath = "/inst/workflow/1_mappings")
```

Comment:
this example's output is slightly changed I think it needs some modification 
![image](https://github.com/user-attachments/assets/2934cd51-5249-485f-8c8a-037157459035)

Reply:
hmmm, yeah. since we are just running the `ae` metric with `RunStep`, we can probably just run the `1_mappings/AE.yaml` workflow to create the data. The name of the lData dfs will need to be updated to match what is in that `AE.yaml`, but that should get you closer to a working example of `RunStep()`

Reply:
I suppose I was thrown off by the `No spec found in workflow. Proceeding without checking data.` statement, but looks like that is being introduced in this current version of `dev` that does not currently exist on `main` . Otherwise this example runs "the same" as the preexisting one on `main` just new messaging, but the outcome is actually the same.

Reply:
Yeah, I didn't tweak the examples at all. Feel free to modify/simplify as needed.